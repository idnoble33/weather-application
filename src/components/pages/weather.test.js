import React from "react";
import * as enzyme from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Weather from "./weather";

enzyme.configure({ adapter: new Adapter() });

describe("Weather", () => {
  const wrapper = shallow(<Weather />);
  it("renders well", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("render the h2 of the weather component", () => {
    expect(wrapper.find("header h2").text()).toContain("Weather");
  });

  test("render a City with text of `Check Weather`", () => {
    expect(wrapper.find(".checkWeather").text()).toBe("Check Weather");
  });
  it("initializes the `state` with an undefined", () => {
    expect(wrapper.state().cityName).toBe(undefined);
  });
});
