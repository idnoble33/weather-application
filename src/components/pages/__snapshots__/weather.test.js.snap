// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather renders well 1`] = `ShallowWrapper {}`;
